org.eclipse.winery.model.threatmodeling/src/test/java/org/eclipse/winery/model/threatmodeling/ThreatModelingUtilsWithGitBackendRepositoryTest.java

```diff
@@ -0,0 +1,181 @@
+
+        QName base = new QName(namespace, "BASE");
+        QName otherBase = new QName(namespace, "OTHER_BASE");
+        QName childA = new QName(namespace, "CHILD_A");
+        QName childB = new QName(namespace, "CHILD_B");
+
+        TNodeType baseType = new TNodeType.Builder("BASE")
+            .setDerivedFrom(QName.valueOf(ThreatModelingConstants.SVNF_NODE_TYPE))
+            .build();
+
+        TNodeType otherBaseType = new TNodeType.Builder("BASE")
+            .build();
+
+        TNodeType childAType = new TNodeType.Builder("CHILD_A").setAbstract(TBoolean.YES).setDerivedFrom(base).build();
+        TNodeType childBType = new TNodeType.Builder("CHILD_B").setAbstract(TBoolean.YES).setDerivedFrom(otherBase).build();
+
+        repository.setElement(new NodeTypeId(base), baseType);
+        repository.setElement(new NodeTypeId(otherBase), otherBaseType);
+        repository.setElement(new NodeTypeId(childA), childAType);
+        repository.setElement(new NodeTypeId(childB), childBType);
+
+        ThreatModelingUtils tmu = new ThreatModelingUtils(repository);
+
+        assertTrue(tmu.isOfTypeOrInheritsFromSVNF(base));
+        assertTrue(tmu.isOfTypeOrInheritsFromSVNF(childA));
+        assertFalse(tmu.isOfTypeOrInheritsFromSVNF(otherBase));
+        assertFalse(tmu.isOfTypeOrInheritsFromSVNF(childB));
+    }
+
+    @Test
+    public void checkFindFirstAbstractTypeThatInheritsFromSVNF() throws Exception {
+        this.setRevisionTo("origin/plain");
+
+        String namespace = "http://example.com";
+
+        QName base = new QName(namespace, "base");
+        QName childA = new QName(namespace, "childA");
+        QName childB = new QName(namespace, "childB");
+        QName notDerivedOrAbstract = new QName(namespace, "notDerivedOrAbstract");
+
+        TNodeType baseType = new TNodeType.Builder("base")
+            .setDerivedFrom(QName.valueOf(ThreatModelingConstants.SVNF_NODE_TYPE))
+            .setAbstract(TBoolean.YES)
+            .build();
+
+        TNodeType childAType = new TNodeType.Builder("childA").setDerivedFrom(base).build();
+        TNodeType childBType = new TNodeType.Builder("childB").setDerivedFrom(childA).build();
+
+        TNodeType notDerivedOrAbstractType = new TNodeType.Builder("notDerivedOrAbstract").build();
+
+        repository.setElement(new NodeTypeId(base), baseType);
+        repository.setElement(new NodeTypeId(childA), childAType);
+        repository.setElement(new NodeTypeId(childB), childBType);
+        repository.setElement(new NodeTypeId(notDerivedOrAbstract), notDerivedOrAbstractType);
+
+        ThreatModelingUtils tmu = new ThreatModelingUtils(repository);
+
+        assertEquals(base, tmu.findFirstAbstractType(base));
+        assertEquals(base, tmu.findFirstAbstractType(childA));
+        assertEquals(base, tmu.findFirstAbstractType(childB));
+
+        assertThrows(Exception.class, () -> tmu.findFirstAbstractType(notDerivedOrAbstract));
```

Comment:
You usually separate these different tests into multiple tests to achieve more fine-grained assertions and results.

Reply:
I see no benefit in separating these tests to be honest since childB depends on childA and this is one case...?